backend/src/main/java/com/kinsella/kinsellaBackend/controllers/ProductsController.java

```diff
@@ -29,17 +38,66 @@ public Collection<Products> getProductByID(@PathVariable Long productID) {
         return productsRepository.findByProductID(productID);
     }
 
+    /*
+    @GetMapping({"/{ticketId}/comment/{commentNumber}", ""})
+    public ResponseEntity getTicketComment(@PathVariable(name = "ticketId") Long ticketId, @PathVariable(name = "commentNumber") Long commentNum){
+    */
+
+    @PostMapping(path = "/admin/add/{name}/{price}/{stock}")
+    public String newProduct (@RequestParam("image") MultipartFile file,
+                              @PathVariable(name = "name") String name,
+                              @PathVariable(name = "price") Double price,
+                              @PathVariable(name = "stock") Integer stock) throws IOException{
+
+        // Create product instance
+        Products product = new Products();
+
+        // Create path to send to database
+        String imagePath = "/productImages/" + file.getOriginalFilename();
+
+        // Save image
+        byte[] bytes = file.getBytes();
+        Path path = Paths.get("../frontend/public/productImages/" + file.getOriginalFilename());
+        Files.write(path, bytes);
+
+        // Set Data
+        product.setProductName(name);
+        product.setPrice(price);
+        product.setStock(stock);
+        product.setImage(imagePath);
 
+        // Save product to database
+        productsRepository.save(product);
 
-    @PostMapping(path = "/admin", consumes = {"application/json"})
-    public Products newProduct(@Validated @RequestBody Products products){
-        return productsRepository.save(products);
+        // Save to database
+        return "Success";
     }
 
     //get product by id
-    @PutMapping("/admin/{productID}")
-    public Products updateProduct(@PathVariable Long productID, @Validated @RequestBody Products productRequest) {
-        return productsRepository.findById(productID).map(product -> {
+    @PutMapping("/admin/{id}/{name}/{price}")
```

Comment:
Same here - you can pass all these in the request body.

Reply:
Applied this change 👍